### DISCUSSION TITLE
You have exceeded a secondary rate limit.

### DISCUSSION THREAD

**Discussion by simoesj-lf** [community/community#29275] [Other Feature Feedback, Questions, & Ideas]
Browsing through github web, I have performed a search within a team and got 100 pages of results. I am looking for something specific withing the repos of that team and need to check every page.

However, after a certain point, when I try to change the page, I get "You have exceeded a secondary rate limit.". I wait for a bit and try again but I can only see 1 or 2 more pages before I get the message again.

I linger on some pages, sometimes open files displayed on a certain page, and on some other pages I just click the next button after 5 seconds or so of scanning.

Is there anyway around this?

**Comment by mldyh** [comment]
it seems GitHub need to deal with their own rate-limit https://docs.github.com/en/rest/guides/best-practices-for-integrators#dealing-with-secondary-rate-limits

if you need to do specific search within a repo, maybe https://github.dev/ will make it easier. Just go to repository that you want to search and press `.` it will send you to github dev. Then do search just like in VSCode.
https://docs.github.com/en/codespaces/the-githubdev-web-based-editor

**Reply by EvertGeerling** [reply]
Same problem with 62 repos.
Going back ONCE gave the error already.
Apparently the caching is not functioning correctly.
Seems a bug to me.

**Comment by johnbent** [comment]
This is super-annoying. I run a volunteer non-profit webpage for the preservation of an endangered language. We have a volunteer group who consults about words which have not yet been documented. We used to use debugle for these discussions but debugle closed. No problem; github is awesome and free. But I wrote a script to automatically create issues in github from the debugle issues and I keep hitting the secondary rate limit. It shouldn't be a problem because I only need to create about 100 issues with 5 comments each. But I keep hitting this limit. It wouldn't be a problem if I could query the limit and then sleep accordingly. But when I try to query, all my rates look OK!!! Argh. :(

~/tekinged.com/scripts/debugle> curl -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $GH_OATH" https://api.github.com/rate_limit | grep remaining
"remaining": 4715,
"remaining": 30,
"remaining": 5000,
"remaining": 5000,
"remaining": 100,
"remaining": 1000,
"remaining": 10000,
"remaining": 15000,
"remaining": 100,
"remaining": 4715,
